spec/design.md

```diff
@@ -250,26 +250,31 @@ MyClass <- Class(slot1 := integer,
 
 ### Methods
 
-Assign the method into the generic, as a member of its dispatch table:
+Methods are defined by calling `method<-(generic, signature, method)`:
 
-```{R}
-generic$Class <- function(x) { }
+```
+method(generic, signature) <- function(x, ...) {} 
 ```
 
-In the above, `Class` would be looked up by name in the parent
-frame. A safer and slightly more verbose syntax, which passes the
-actual class object:
+* `generic` is either a function or a string representing the name of a
+  function which is looked up in the calling frame.
+  
+* `signature` is a single class object, list of class objects, or a 
+  character vector. If a character vector is supplied, class objects
+  are searched for in the calling frame.
 
-```{R}
-generic[[classObject]] <- function(x) { }
-```
+`method<-` will perform method validation ensuring that the method is compatible with the generic (i.e. all arguments before `...` have the same names in the same order; if the generic doesn't have `...` all arguments must be same). 
 
-For multiple (nested) dispatch:
-```{R}
-generic[[classObject1]][[classObject2]] <- function(x) { }
-```
+You should only defined a method if you "own" either the generic, or at least one of the classes in the signature. This can not be enforced programmatically (because there's no straightforward defintion of ownership), but will be recommended as best practice.
```

Comment:
Perhaps it would be simplest to just say that the documentation will explain the risks involved when defining a method for a third party generic and class. For example, it is common practice now for generics to be defined by a "stub" package, like the MatrixGenerics package. Ideally all of the "class owners" would implement methods for them, but in practice some other community member may implement compatibility. Being too blunt in our documentation may discourage what would otherwise be a useful contribution. Ultimately, this is more of a "community norm" than a technical consideration.